src/System.Diagnostics.Debug/tests/DebugTestsNoListeners.cs

```diff
@@ -0,0 +1,239 @@
+            VerifyLogged(() => Debug.WriteIf(false, 5, "category"), "");
+
+            VerifyLogged(() => Debug.WriteIf(true, "logged"), "logged");
+            VerifyLogged(() => Debug.WriteIf(false, "logged"), "");
+
+            VerifyLogged(() => Debug.WriteIf(true, "logged", "category"), "category: logged");
+            VerifyLogged(() => Debug.WriteIf(false, "logged", "category"), "");
+
+            GoToNextLine();
+        }
+
+        [Fact]
+        public void WriteLineIf()
+        {
+            VerifyLogged(() => Debug.WriteLineIf(true, 5), "5" + Environment.NewLine);
+            VerifyLogged(() => Debug.WriteLineIf(false, 5), "");
+
+            VerifyLogged(() => Debug.WriteLineIf(true, 5, "category"), "category: 5" + Environment.NewLine);
+            VerifyLogged(() => Debug.WriteLineIf(false, 5, "category"), "");
+
+            VerifyLogged(() => Debug.WriteLineIf(true, "logged"), "logged" + Environment.NewLine);
+            VerifyLogged(() => Debug.WriteLineIf(false, "logged"), "");
+
+            VerifyLogged(() => Debug.WriteLineIf(true, "logged", "category"), "category: logged" + Environment.NewLine);
+            VerifyLogged(() => Debug.WriteLineIf(false, "logged", "category"), "");     
+        }
+
+        [Theory]
+        [InlineData(-1, 0)]
+        [InlineData(0, 0)]
+        [InlineData(1, 1)]
+        [InlineData(2, 2)]
+        [InlineData(3, 3)]
+        [InlineData(4, 4)]
+        public void IndentSize_Set_GetReturnsExpected(int indentSize, int expectedIndentSize)
+        {
+            Debug.IndentLevel = 0;
+
+            Debug.IndentSize = indentSize;
+            Assert.Equal(expectedIndentSize, Debug.IndentSize);
+            VerifyLogged(() => Debug.WriteLine("pizza"), "pizza" + Environment.NewLine);
+
+            // Indent once.
+            Debug.Indent();
+            string expectedIndentOnce = new string(' ', expectedIndentSize);
+            VerifyLogged(() => Debug.WriteLine("pizza"), expectedIndentOnce + "pizza" + Environment.NewLine);
+
+            // Indent again.
+            Debug.Indent();
+            string expectedIndentTwice = new string(' ', expectedIndentSize * 2);
+            VerifyLogged(() => Debug.WriteLine("pizza"), expectedIndentTwice + "pizza" + Environment.NewLine);
+
+            // Unindent.
+            Debug.Unindent();
+            Debug.Unindent();
+        }
+
+        [Theory]
+        [InlineData(-1, 0)]
+        [InlineData(0, 0)]
+        [InlineData(1, 1)]
+        public void IndentLevel_Set_GetReturnsExpected(int indentLevel, int expectedIndentLevel)
+        {
+            Debug.IndentLevel = indentLevel;
+            Assert.Equal(expectedIndentLevel, Debug.IndentLevel);
+            string expectedIndentOnce = new string(' ', expectedIndentLevel * Debug.IndentSize);
+            VerifyLogged(() => Debug.WriteLine("pizza"), expectedIndentOnce + "pizza" + Environment.NewLine);
+
+            // Indent once.
+            Debug.Indent();
+#if DEBUG
```

Comment:
It would be better to define DEBUG at the top of the file instead, so that even the release build flavor is testing the APIs.